websub-ballerina/src/websub_generic_integration_tests/tests/integration-tests/subscriber/test_subscriber_startup.bal

```diff
@@ -0,0 +1,126 @@
+        websub:Hub|websub:HubStartedUpError|websub:HubStartupError res =
+            websub:startHub(lis0, "/websub", "/hub", "/pub", publicUrl = "https://localhost:23391");
+
+        if (res is websub:Hub) {
+            if (res.publishUrl != "https://localhost:23391/websub/pub" ||
+                    res.subscriptionUrl != "https://localhost:23391/websub/hub") {
+                return caller->respond("invalid publishUrl and subscriptionUrl");
+            }
+        } else {
+            io:println(res);
+            return caller->respond("hub(lis0) start up error");
+        }
+
+        // testHubStartUpWhenStarted
+        websub:Hub|websub:HubStartedUpError|websub:HubStartupError res2 = websub:startHub(lis1);
+
+        if !(res2 is websub:HubStartedUpError) || res2.startedUpHub !== res {
+            return caller->respond("seperate hub(lis1) has started");
+        }
+
+        // testHubShutdownAndStart
+        websub:Hub hub = <websub:Hub> res;
+        error? err = hub.stop();
+        if (err is error) {
+            io:println(err);
+            return caller->respond("hub(lis0) stop error");
+        }
+
+        http:Listener lis2 = new (23393);
+        res2 = websub:startHub(lis2);
+        if res2 is websub:Hub {
+            string responseMsg = res2.publishUrl == "http://localhost:23393/publish" && res2.subscriptionUrl ==
+                            "http://localhost:23393/" ? "hub(lis2) start successfully" :
+                            "incorrect hub(lis2) has started";
+            err = res2.stop();
+            return caller->respond(responseMsg);
+        }
+        return caller->respond("hub(lis2) start up error");
+    }
+
+    resource function testPublisherAndSubscriptionInvalidSameResourcePath(http:Caller caller, http:Request request)
+                                                                                                    returns error? {
+        http:Listener lis = new (23394);
+        websub:Hub|websub:HubStartedUpError|websub:HubStartupError res =
+            websub:startHub(lis, "/websub", "/hub", "/hub");
+
+        var err = lis.__gracefulStop();
+
+        if (res is websub:HubStartupError) {
+            return caller->respond(res.message());
+        }
+        return caller->respond("Unexpected result");
+    }
+}
+
+@test:Config {
+    dependsOn: ["testTopicRedirectFoundAndHubPermanentRedirect"]
+}
+function testMultipleSubscribersStartUpInSamePort() {
+    http:Client clientEndpoint = new ("http://0.0.0.0:23386");
+    var response = clientEndpoint->get("/subscriber/start");
+    HttpResponseDetails responseDetails = fetchHttpResponse(response);
+
+    string expectedResponseMsg = "failed to start server connector '0.0.0.0:23387': Address already in use";
+    test:assertEquals(responseDetails.statusCode, http:STATUS_OK, msg = "Response code mismatched");
+    test:assertEquals(responseDetails.responseMessage, expectedResponseMsg, msg = "Mismatched");
+
+}
```

Comment:
```suggestion
}

```

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-websub/pull/1/commits/7defc19195f199d3899556e82f9a35eb1ebf4bd5